### ISSUE TITLE
M:6/8 – animation plays too fast

### ISSUE THREAD

**Issue by tobyspark** [paulrosen/abcjs#155]
For tunes with M:6/8 (and possibly other meters), the animation is not in sync with audio

I am using `abcjs_midi_5.1.2-min.js` from the repo and have `npm run build` myself from HEAD (185bcf78e2021967cd0f404d16f0d940a8fadcf4 Correct the speed of animation in 3/8 meter – that last commit looks to have tweaked relevant code).

With an ABCJS.Editor instantiated like so –
```
new ABCJS.Editor("abc-" + i, { 
            canvas_id: "staff-" + i,
            generate_midi: true,
            midi_id:"midi-" + i,
            midi_download_id: "midi-download-" + i,
            abcjsParams: {
                generateInline: true,
                generateDownload: true,
                downloadLabel:"Download MIDI",
                paddingleft:0,
                paddingright:0,
                responsive: "resize",
                animate: {
                    listener: animateCallback, 
                    target: "staff-" + i, 
                    qpm: 120
                },
            }
        });
```

This follows the score (i.e. animation) faster than the audio playback.
```
M:6/8
K:Cmaj
eccccc|eccccc|eccccc|eccccc|
```

Whereas this equivalent ABC in 4/4 follows the score in time with the audio playback, and that audio playback is the same speed as the above.
```
M:4/4
K:Cmaj
ecccccec|cccceccc|cceccccc|
```

Adding `Q:1/4=120` to the 6/8 tune results in correct score following.

i. That looks like a bug, adding the Q line should be unnecessary.
ii. I don't understand why the animate dict on creating the editor has a qpm attribute. Shouldn't that be picked up from the ABC and match the audio playback?

**Comment by tobyspark** [comment]
The above commit in my fork appears to fix this issue. Tested on 3/8, 6/8, 9/8, 12/8 meters.

Suspect there is more going on in the codebase though – there seems to be semi-duplicated 'set bpm' type functions, and the original code here, setting beat length to 3/8 suggests compound time was being handled slightly differently (as it should be, to my understanding).

**Comment by tobyspark** [comment]
Thanks! I can do a PR if that makes any difference.

(On qpm, I later came to the root of my confusion: the documentation makes it sound like it's a required attribute, whereas leaving it out will instead pick up from the abc and default to a hard-coded figure).

(Plus: on tempo boxes, oh no they can't! I haven't raised a PR for this yet, but here's the line
https://github.com/tobyspark/abcjs/commit/d78683f2fd4cecc609117bbfed3442bea66009a8)

**Comment by paulrosen** [comment]
Will be fixed in 5.3.0.
